### PR TITLE
chore: deprecate remote

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
Ref #21408.

This adds a new deprecation warning message when `require('electron').remote` is encountered.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Deprecated the `remote` module. It is replaced by [`@electron/remote`](https://github.com/electron/remote).

### CODE REVIEW

File: docs/breaking-changes.md [hunk 1]
```diff
@@ -50,6 +50,25 @@ If your crash ingestion server does not support compressed payloads, you can
 turn off compression by specifying `{ compress: false }` in the crash reporter
 options.
 
+### Deprecated: `remote` module
+
+The `remote` module is deprecated in Electron 12, and will be removed in
```
Comment (MarshallOfSound) [inline docs/breaking-changes.md:55]: Can we utilize this message to further indicate that you shouldn't just replace and forget.  Let's use it as another chance to educate why using `remote` is bad.

File: lib/renderer/api/remote.ts [hunk 2]
```diff
@@ -3,9 +3,12 @@ import { isPromise, isSerializableObject, serialize, deserialize } from '../../c
 import { MetaTypeFromRenderer, ObjectMember, ObjProtoDescriptor, MetaType } from '../../common/remote/types';
 import { ipcRendererInternal } from '../ipc-renderer-internal';
 import type { BrowserWindow, WebContents } from 'electron/main';
+import deprecate from '@electron/internal/common/api/deprecate';
 import { browserModuleNames } from '@electron/internal/browser/api/module-names';
 import { commonModuleList } from '@electron/internal/common/api/module-list';
 
+deprecate.log('The remote module is deprecated. Use https://github.com/electron/remote instead.');
```
Comment (MarshallOfSound) [inline lib/renderer/api/remote.ts:10]: Same here

**Review by jkleinsc — CHANGES_REQUESTED** [review]
We should also update this doc to note that remote is deprecated: 
https://www.electronjs.org/docs/api/remote

File: docs/breaking-changes.md (hunk 1 above)
Comment (nornagon) [reply docs/breaking-changes.md:55]: The @electron/remote repository has a big warning at the top. I think that's enough.

**Comment by nornagon** [comment]
@jkleinsc that's already been done in master. #24941

**Review by jkleinsc — APPROVED** [review]
Doh! @nornagon I missed that the doc had been updated.

**Review by codebytere — APPROVED** [review]

**Review by zcbenz — APPROVED** [review]

**Pull request merged**